src/Logger.php

```diff
@@ -4,58 +4,114 @@
     }
 
-    public function error(string $message): void
+    /**
+     * Log an emergency message to the logs.
+     *
+     * @param TMessage $message
+     * @param TContext $context
+     */
+    public function error(string|\Stringable $message, array $context = []): void
     {
-        try {
-            $this->rpc->call('Error', $message);
-        } catch (ServiceException $e) {
-            $this->handleError($e);
-        }
+        $this->send(LogLevel::Error, $message, $context);
     }
 
-    public function warning(string $message): void
+    /**
+     * Log a warning message to the logs.
+     *
+     * @param TMessage $message
+     * @param TContext $context
+     */
+    public function warning(string|\Stringable $message, array $context = []): void
     {
-        try {
-            $this->rpc->call('Warning', $message);
-        } catch (ServiceException $e) {
-            $this->handleError($e);
-        }
+        $this->send(LogLevel::Warning, $message, $context);
     }
 
-    public function info(string $message): void
+    /**
+     * Log an informational message to the logs.
+     *
+     * @param TMessage $message
+     * @param TContext $context
+     */
+    public function info(string $message, array $context = []): void
     {
-        try {
-            $this->rpc->call('Info', $message);
-        } catch (ServiceException $e) {
-            $this->handleError($e);
-        }
+        $this->send(LogLevel::Info, $message, $context);
     }
 
-    public function debug(string $message): void
+    /**
+     * Log a debug message to the logs.
+     *
+     * @param TMessage $message
+     * @param TContext $context
+     */
+    public function debug(string|\Stringable $message, array $context = []): void
     {
-        try {
-            $this->rpc->call('Debug', $message);
-        } catch (ServiceException $e) {
-            $this->handleError($e);
-        }
+        $this->send(LogLevel::Debug, $message, $context);
+    }
+
+    /**
+     * Log a message to the logs.
+     *
+     * @param TMessage $message
+     * @param TContext $context
+     */
+    public function log(string|\Stringable $message, array $context = []): void
+    {
+        $this->send(LogLevel::Log, $message, $context);
     }
 
-    public function log(string $message): void
+    /**
+     * @param TContext $context
+     * @throws LoggerException
+     */
+    private function send(LogLevel $level, string|\Stringable $message, array $context = []): void
     {
         try {
-            $this->rpc->call('Log', $message);
+            if ($context === []) {
+                $this->rpc->call($level->name, (string)$message);
+            } else {
+                $attrs = [];
+
+                foreach ($context as $key => $value) {
+                    $attrs[] = new LogAttrs(data: [
+                        'key' => $key,
+                        'value' => \is_string($value) ? $value : \json_encode($value),
```

Comment:
1. I think it might be good to check Stringable there and cast to string if possible
2. Need to add more options into `json_encode` and try/catch block